src/Adapters/Storage/Postgres/PostgresStorageAdapter.js

```diff
@@ -1797,7 +1797,12 @@ export class PostgresStorageAdapter implements StorageAdapter {
         if (key === 'ACL') {
           memo.push('_rperm');
           memo.push('_wperm');
-        } else if (key.length > 0) {
+        } else if (
+          key.length > 0 &&
+          // Remove selected field not referenced in the schema
+          // Relation is not a column in postgres
+          ((schema.fields[key] && schema.fields[key].type !== 'Relation') || key === '$score')
```

Comment:
And can you explain what is the reason for this restrictive condition? Before it was just checking the key length, now it is checking whether the key more carefully. I that something required for this PR?

Reply:
@Moumouls I'd like to get this merged; is this restriction necessary for this PR? It's the only thing why I am hesitant to merge, because this restriction is quite a change from previously only checking the length.

Reply:
> is this restriction necessary for this PR

Yes otherwise postgres will fail on many GraphQL queries.

> because this restriction is quite a change from previously only checking the length.

Currently the change seems to not to fail other tests, and this fix deeply fix many graphql issues for parse developers using postgres.

Honestly I think we are good 🙂

Reply:
ok, let's merge it then